engines/bladerunner/fog.cpp

```diff
@@ -135,12 +136,14 @@ void FogSphere::calculateCoeficient(Vector3 position, Vector3 viewPosition, floa
 	Vector3 rayDirection = (rayDestination - rayOrigin).normalize();
 
 	float b = Vector3::dot(rayDirection, rayOrigin);
-	float c = Vector3::dot(rayOrigin, rayOrigin) - (_radius * _radius);
+	float c = Vector3::dot(rayOrigin, rayOrigin) - _radius_e2;
 	float d = b * b - c;
 
 	if (d >= 0.0f) { // there is an interstection between ray and the sphere
-		Vector3 intersection1 = rayOrigin + (-b - sqrt(d)) * rayDirection;
-		Vector3 intersection2 = rayOrigin + (-b + sqrt(d)) * rayDirection;
+		d = sqrt(d);
```

Comment:
I guess this is safe in this situation, but it may cause potential issues in the future, since the value of d whose scope is outside the if clause, is changed here to its square root. I'd prefer to use a different variable, ie. "d_sqrt" with scope within the if clause